app/screens/edit_profile/components/select_field.tsx

```diff
@@ -0,0 +1,121 @@
+    return {
+        container: {
+            marginVertical: 8,
+            paddingHorizontal: 20,
+            width: '100%',
+            backgroundColor: theme.centerChannelBg,
+        },
+        tabletContainer: {
+            paddingHorizontal: 42,
+        },
+    };
+});
+
+const SelectField = ({
+    fieldKey,
+    label,
+    value,
+    options,
+    isDisabled = false,
+    onValueChange,
+    onFocusNextField,
+    testID,
+    isOptional = false,
+    isMultiselect = false,
+}: SelectFieldProps) => {
+    const theme = useTheme();
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const styles = getStyleSheet(theme);
+
+    let formattedLabel = label;
+    if (isOptional) {
+        formattedLabel = intl.formatMessage(
+            {
+                id: 'select_field.optional',
+                defaultMessage: '{labelName} (optional)',
+            },
+            {labelName: label},
+        );
+    }
+
+    // Convert stored value to selected format for AutocompleteSelector
+    const selectedValue = useMemo(() => {
+        if (!value) {
+            return isMultiselect ? [] : undefined;
+        }
+
+        const selectedIds = getSelectedOptionIds(value, isMultiselect ? 'multiselect' : 'select');
+
+        if (isMultiselect) {
+            return selectedIds;
+        }
+
+        return selectedIds[0] || undefined;
+    }, [value, isMultiselect]);
+
+    const handleSelect = useCallback((newSelection?: SelectedDialogOption) => {
+        if (!newSelection) {
+            onValueChange(fieldKey, '');
+            return;
+        }
+
+        if (Array.isArray(newSelection)) {
+            // Multiselect: convert array of selections to JSON string of IDs
+            const selectedIds = newSelection.map((option) => option.value);
+            onValueChange(fieldKey, JSON.stringify(selectedIds));
+        } else {
+            // Single select: store the option ID
+            onValueChange(fieldKey, newSelection.value);
+        }
+
+        // Focus next field after selection
+        onFocusNextField(fieldKey);
+    }, [fieldKey, onValueChange, onFocusNextField]);
+
+    const containerStyle = useMemo(() => [
+        styles.container,
+        isTablet && styles.tabletContainer,
+    ], [isTablet, styles]);
+
+    return (
+        <View
+            style={containerStyle}
+            testID={testID}
+        >
+            <AutocompleteSelector
+                label={formattedLabel}
+                options={options}
+                selected={selectedValue}
+                onSelected={handleSelect}
+                disabled={isDisabled}
+                isMultiselect={isMultiselect}
+                testID={`${testID}.selector`}
+                location={Screens.EDIT_PROFILE}
+                placeholder={intl.formatMessage({
+                    id: 'mobile.action_menu.select',
+                    defaultMessage: 'Select an option',
```

Comment:
UX question, for multi-select. Should we keep the defaultMessage as `Select an option`? It sounds singular, which might be confusing in a multi-select context.